txpool/account.go

```diff
@@ -135,9 +135,46 @@ func (m *accountsMap) allTxs(includeEnqueued bool) (
 	return
 }
 
+// resetWithNonce updates all accounts with the new nonce and clears any stale transactions.
+// May signal a promotion request if the account is eligible after pruning.
+func (m *accountsMap) resetWithNonce(newNonces map[types.Address]uint64, promoteCh chan<- promoteRequest) (
+	allPrunedPromoted,
+	allPrunedEnqueued []*types.Transaction,
+) {
+	//	prune each account with the new nonce
+	m.Range(func(key, value interface{}) bool {
+		addr, _ := key.(types.Address) //nolint:forcetypeassert, nolintlint
+		account, _ := value.(*account) //nolint:forcetypeassert, nolintlint
+
+		newNonce, ok := newNonces[addr]
+		if !ok {
+			// no updates for this account
+			return true
+		}
+
+		//	prune stale txs
+		prunedPromoted, prunedEnqueued := account.reset(newNonce, promoteCh)
+
+		//	update result
+		allPrunedPromoted = append(
+			allPrunedPromoted,
+			prunedPromoted...,
+		)
+
+		allPrunedEnqueued = append(
+			allPrunedEnqueued,
+			prunedEnqueued...,
+		)
+
+		return true
+	})
```

Comment:
Why do you need to scan all accounts? I think the only thing you need to do is to scan newNonces. Number of accounts is growing, while number of newNonces keeps below a certain number.

Reply:
Sorry, I don't get it. I think sync.Map::Range won't block from other accessing to the same key.

Reply:
Yes, you're right - it won't block other goroutines from the same entry/

It boils down to which of these 2 is faster (pseudocode below):

`Range` approach:
```
m.Range(key, value) {
    addr = key

    if no_new_updates(addr) == true {
        continue
    }

    resetAccount(addr)
}
```

or

regular `for` loop:
```
for addr in newNonces {
    if !account_exists(addr) {
        continue
    }

    resetAccount(addr)
}
```

The actual locking is performed in `resetAccount()`.

Reply:
I think scanning all accounts takes much computation cost. second approach is better

Reply:
@zivkovicmilos What do you think?

Reply:
I agree with @Kourin1996 on this one, purely because the number of accounts who are gonna se updates is usually going to be less than the number of all accounts currently loaded in memory - that's why I think #2 is better, since you're only ranging on accounts that need to be ranged upon.

The performance boost is I believe minimal for small data sets, you'd need to really stress it to see any kind of gains, I'm all for just going with the second approach and dropping the benchmark 👍

Reply:
Sound good to me 👍 

[Refactored](https://github.com/0xPolygon/polygon-edge/pull/459/commits/45284cc1fd33b30f9de52ad875f5a8eca48d6a6e)